### PR TITLE
Separate input moving to utils file

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77187

In preparation for FSDP `device_id` support which will move GPU inputs to the device specified, move these codes to a utils file.

TODO: type the code: https://github.com/pytorch/pytorch/issues/77190

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Related request on implementing a recursive utility of moving to target device into core `torch.to` function: https://github.com/pytorch/pytorch/issues/69431

**Review by awgu — APPROVED** [review]
LGTM!

**Comment by awgu** [comment]
Maybe check the lint errors:
```
See https://www.flake8rules.com/rules/E306.html
2022-05-10T19:25:22.3542916Z ##[warning]'collections.abc' imported but unused
See https://www.flake8rules.com/rules/F401.html
2022-05-10T19:25:22.3544924Z ##[warning]'._functions._get_stream' imported but unused
```

File: torch/nn/parallel/distributed.py [hunk 1]
```diff
@@ -36,9 +36,8 @@
 from torch._utils import _get_device_index
 
 from ..modules import Module
-from ._functions import _get_stream
 from ._replicated_tensor_ddp_utils import _ddp_with_replicated_tensor_enabled
-from .scatter_gather import gather, is_namedtuple, scatter_kwargs
```
Comment (rohan-varma) [inline torch/nn/parallel/distributed.py:41]: actually we should not remove is_namedtuple due to bc

**Pull request closed without merging**